src/sb_pointcloud_processing/include/BallExtractorNode.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Created By: Min Gyo Kim
+ * Created On: September 29, 2018
+ * Description: Header file for Ball Extractor Node
+ *  This node subscribes to a point cloud topic and expects the topic to contain
+ * a point cloud
+ *  that is filtered to contain roughly only the points that represent the
+ * tennis ball.
+ *  Then the node performs DBSCAN on the Y-Z plane, and calculates a rough
+ * estimate of the
+ *  center of the tennis ball, and publishes the point.
+ */
+
+#ifndef BALL_EXTRACTOR_IGVC_NODE_H
+#define BALL_EXTRACTOR_IGVC_NODE_H
+
+#include <DBSCAN.h>
+#include <RvizUtils.h>
+#include <iostream>
+#include <math.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <ros/console.h>
+#include <ros/ros.h>
+#include <sb_utils.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <string>
+
+class BallExtractorNode {
+  public:
+    /*
+     * @clusters stores the output from DBSCAN
+     */
+    std::vector<pcl::PointCloud<pcl::PointXYZ>> clusters;
+
+    BallExtractorNode(int argc, char** argv, std::string node_name);
+
+    // main entry function
+    void extractBall();
+
+    /*
+     * @clusters: input clusters
+     * @cluster_points: vector to store converted cluster points
+     * @colors: vector to store color for each cluster point
+     * This function takes in @clusters, assigns a different color
+     * for each cluster, and converts each point to geometry_msgs::Point.
+     * It pushes the points and colors to the back of @cluster_points
+     * and @colors.
```

Comment:
Funky formatting here, we should re-format this into a proper javadoc comment, ie.
```
/**
 * One line description here
 * 
 * More through description here, possibly spanning over multiple 
 * lines
 *
 * @param some_param This is a parameter that we use for things
 * @return Some function of the parameter
 */
```